programs/bpf_loader/src/lib.rs

```diff
@@ -667,51 +795,61 @@ fn process_loader_upgradeable_instruction(
             }
 
             let buffer_data_offset = UpgradeableLoaderState::buffer_data_offset()?;
-            let buffer_data_len = buffer.data_len()?.saturating_sub(buffer_data_offset);
+            let buffer_data_len = buffer.get_data().len().saturating_sub(buffer_data_offset);
+            let programdata = instruction_context.try_borrow_instruction_account(
+                transaction_context,
+                upgradeable_ins_acc_idx::Upgrade::ProgramData as usize,
+            )?;
             let programdata_data_offset = UpgradeableLoaderState::programdata_data_offset()?;
-            let programdata_balance_required = 1.max(rent.minimum_balance(programdata.data_len()?));
-
-            if buffer.data_len()? < UpgradeableLoaderState::buffer_data_offset()?
+            let programdata_balance_required =
+                1.max(rent.minimum_balance(programdata.get_data().len()));
+            if buffer.get_data().len() < UpgradeableLoaderState::buffer_data_offset()?
                 || buffer_data_len == 0
             {
                 ic_logger_msg!(log_collector, "Buffer account too small");
                 return Err(InstructionError::InvalidAccountData);
             }
 
             // Verify ProgramData account
-
-            if programdata.data_len()? < UpgradeableLoaderState::programdata_len(buffer_data_len)? {
+            if programdata.get_data().len()
+                < UpgradeableLoaderState::programdata_len(buffer_data_len)?
+            {
                 ic_logger_msg!(log_collector, "ProgramData account not large enough");
                 return Err(InstructionError::AccountDataTooSmall);
             }
-            if programdata.lamports()? + buffer.lamports()? < programdata_balance_required {
+            if programdata.get_lamports() + buffer.get_lamports() < programdata_balance_required {
```

Comment:
Be helpful if these name changes came in a different PR (before) to reduce the noise when trying to review and find order/behavior differences.

Reply:
This is not a rename of methods, it uses `BorrowedAccount` instead of `KeyedAccount` now.

Reply:
Was thinking that borrowed account export`lamports()` and then once all the rework is complete switch the name over.  That way the reworking of accounts/error order is easier to grok in this pr